vision/convmixer_cifar10/convmixer.jl

```diff
@@ -0,0 +1,131 @@
+using Flux, MLDatasets
+using Flux: onehotbatch, onecold, DataLoader, Optimiser
+using BSON:@save,@load
+
+
+function ConvMixer(in_channels, kernel_size, patch_size, dim, depth, N_classes)
+    f = Chain(
+            Conv((patch_size, patch_size), in_channels=>dim, gelu; stride=patch_size),
+            BatchNorm(dim),
+            [
+                Chain(
+                    SkipConnection(Chain(Conv((kernel_size,kernel_size), dim=>dim, gelu; pad=SamePad(), groups=dim), BatchNorm(dim)), +),
+                    Chain(Conv((1,1), dim=>dim, gelu), BatchNorm(dim))
+                ) 
+                for i in 1:depth
+            ]...,
+            AdaptiveMeanPool((1,1)),
+            flatten,
+            Dense(dim,N_classes)
+        )
+    return f
+end
+
+function get_data(batchsize; dataset = MLDatasets.CIFAR10, idxs = nothing)
+    """
+    idxs=nothing gives the full dataset, otherwise (for testing purposes) only the 1:idxs elements of the train set are given.
+    """
+    ENV["DATADEPS_ALWAYS_ACCEPT"] = "true" 
+
+    # Loading Dataset
+    if idxs==nothing
+        xtrain, ytrain = dataset.traindata(Float32)
+        xtest, ytest = dataset.testdata(Float32)
+	else
+        xtrain, ytrain = dataset.traindata(Float32, 1:idxs)
+        xtest, ytest = dataset.testdata(Float32, 1:Int(idxs/10))
+    end
+
+    # Reshape Data to comply to Julia's (width, height, channels, batch_size) convention in case there are only 1 channel (eg MNIST)
+    if ndims(xtrain)==3
+        w = size(xtrain)[1]
+        xtrain = reshape(xtrain, (w,w,1,:))
+        xtest = reshape(xtest, (w,w,1,:))
+    end
+    
+    ytrain, ytest = onehotbatch(ytrain, 0:9), onehotbatch(ytest, 0:9)
+
+    train_loader = DataLoader((xtrain, ytrain), batchsize=batchsize, shuffle=true)
+    test_loader = DataLoader((xtest, ytest), batchsize=batchsize)
+
+    return train_loader, test_loader
+end
+
+function ℓ(dataloader, model, device)
```

Comment:
Beat to call this `loss` for simplicity. Rather than taking in `device` could we not generically use `gpu`? It's free if there is no GPU available and there's little reason to not take the free speed otherwise.